### PR TITLE
[dashboard-oop] PR feedback

### PR DESCRIPTION
**Pull request by JamesNK**
Feedback on https://github.com/dotnet/aspire/pull/1774
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1805)

### CODE REVIEW

**Review by mitchdenny — APPROVED** [review]
Looks good. I was just working in the wait loop myself and your solution is similar. I noticed a bug that I had where i was using cancellationToken and not cts.Token for the linked cancellation token.

**Comment by JamesNK** [comment]
> Looks good. I was just working in the wait loop myself and your solution is similar. I noticed a bug that I had where i was using cancellationToken and not cts.Token for the linked cancellation token.

Now that you mention it, I took a closer look at the method. I've updated the method to only show the timeout exception message if there is a timeout and not for other cancellation.

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 1]
```diff
@@ -178,31 +186,40 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
 
     private async Task WaitForHttpSuccessOrThrow(string url, TimeSpan timeout, CancellationToken cancellationToken = default)
     {
-        using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-        cts.CancelAfter(timeout);
+        using var timeoutCts = new CancellationTokenSource();
+        timeoutCts.CancelAfter(timeout);
```
Comment (drewnoakes) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:190]: Can be made slightly simpler.

```suggestion
        using var timeoutCts = new CancellationTokenSource(timeout);
```

File: src/Shared/KnownFormats.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire;
+
+public static class KnownFormats
```
Comment (drewnoakes) [inline src/Shared/KnownFormats.cs:6]: Internal, in case we link to multiple projects and introduce conflicts.

```suggestion
internal static class KnownFormats
```

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 3]
```diff
@@ -123,7 +123,7 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
             WorkingDirectory = dashboardWorkingDirectory
         };
 
-        if (StringComparer.OrdinalIgnoreCase.Equals(".dll", Path.GetExtension(fullyQualifiedDashboardPath)))
+        if (string.Equals(".dll", Path.GetExtension(fullyQualifiedDashboardPath), StringComparison.OrdinalIgnoreCase))
```
Comment (drewnoakes) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:126]: Why this change? Seems equivalent but more verbose.

**Review by drewnoakes — APPROVED** [review]

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs (hunk 3 above)
Comment (JamesNK) [reply src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:126]: I use them like this answer describes: https://stackoverflow.com/a/8918112
